### PR TITLE
refactor(animations): support browser animation rendering

### PR DESCRIPTION
**Pull request by matsko**

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -5,6 +5,8 @@ set -e -o pipefail
 cd `dirname $0`
 
 PACKAGES=(core
+  animation
+  animation-browser
```
Comment (mhevery) [inline build.sh]: Animation browser needs to be a sub folder in animation.

File: modules/@angular/animation-browser/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (mhevery) [inline modules/@angular/animation-browser/package.json]: Jason can help you to move this to sub folder

File: modules/@angular/animation-browser/src/dsl/animation.ts [hunk 3]
```diff
@@ -34,26 +32,22 @@ export class Animation {
...
-        <StyleData>destinationStyles;
+    const start = Array.isArray(startingStyles) ? normalizeStyles(startingStyles) : <StyleData>startingStyles;
+    const dest = Array.isArray(destinationStyles) ? normalizeStyles(destinationStyles) : <StyleData>destinationStyles;
     return buildAnimationKeyframes(this._animationAst, start, dest);
   }
 
   // this is only used for development demo purposes for now
   private create(
-      injector: Injector, element: any, startingStyles: StyleData = {},
+      injector: any, element: any, startingStyles: StyleData = {},
```
Comment (mhevery) [inline modules/@angular/animation-browser/src/dsl/animation.ts]: You should not be expecting an injector here. Even if you pass any. Can this be solved in a different way?

File: modules/@angular/animation-browser/src/engine/animation_engine.ts [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {AnimationPlayer, NoOpAnimationPlayer, AnimationGroupPlayer, AnimationTriggerMetadata, StyleData, AnimationEvent} from '@angular/animation';
+import {AnimationTimelineInstruction} from '../dsl/animation_timeline_instruction';
+import {AnimationTransitionInstruction} from '../dsl/animation_transition_instruction';
+import {AnimationStyleNormalizer} from '../dsl/style_normalization/animation_style_normalizer';
+import {AnimationDriver} from './animation_driver';
+import {AnimationTrigger, buildTrigger} from "../dsl/animation_trigger";
+
+export declare type AnimationPlayerTuple = {
```
Comment (mhevery) [inline modules/@angular/animation-browser/src/engine/animation_engine.ts]: Theses should be interfaces not declare type.

File: modules/@angular/animation-browser/src/dsl/animation.ts (hunk 3 above)
Comment (tbosch) [inline modules/@angular/animation-browser/src/dsl/animation.ts]: This should fail the typescript compile actually. @matsko can you check your `package.json`, it should not include anything from angular...

File: modules/@angular/animation-browser/src/engine/animation_engine.ts [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {AnimationPlayer, NoOpAnimationPlayer, AnimationGroupPlayer, AnimationTriggerMetadata, StyleData, AnimationEvent} from '@angular/animation';
+import {AnimationTimelineInstruction} from '../dsl/animation_timeline_instruction';
+import {AnimationTransitionInstruction} from '../dsl/animation_transition_instruction';
+import {AnimationStyleNormalizer} from '../dsl/style_normalization/animation_style_normalizer';
+import {AnimationDriver} from './animation_driver';
+import {AnimationTrigger, buildTrigger} from "../dsl/animation_trigger";
```
Comment (DzmitryShylovich) [inline modules/@angular/animation-browser/src/engine/animation_engine.ts]: single quotes for consistency :)

**Review by IgorMinar — APPROVED** [review]

**Pull request merged**